components/edit_post/edit_post.tsx

```diff
@@ -0,0 +1,494 @@
+            const shouldScrollIntoView = shouldScroll(elementCR, parentCR);
+
+            if (shouldScrollIntoView) {
+                const scrollOptions: boolean|ScrollIntoViewOptions = isFirefox() ? {behavior: 'smooth', block: 'nearest'} : false;
+                wrapperRef?.current?.scrollIntoView(scrollOptions);
+            }
+        }
+
+        textboxRef?.current?.focus();
+    }, []);
+
+    useEffect(() => {
+        const handlePaste = (e: ClipboardEvent) => {
+            if (
+                !e.clipboardData ||
+                !e.clipboardData.items ||
+                !rest.canEditPost ||
+                (e.target as HTMLTextAreaElement).id !== 'edit_textbox'
+            ) {
+                return;
+            }
+
+            const {clipboardData} = e;
+            const table = getTable(clipboardData);
+
+            if (!table) {
+                return;
+            }
+
+            e.preventDefault();
+
+            let message = editText;
+            let newCaretPosition = caretPosition;
+
+            if (table && isGitHubCodeBlock(table.className)) {
+                const {formattedMessage, formattedCodeBlock} = formatGithubCodePaste(
+                    caretPosition,
+                    message,
+                    clipboardData,
+                );
+                newCaretPosition = caretPosition + formattedCodeBlock.length;
+                message = formattedMessage;
+            } else if (table) {
+                message = formatMarkdownTableMessage(table, editText.trim(), newCaretPosition);
+                newCaretPosition = message.length - (editText.length - newCaretPosition);
+            }
+
+            setEditText(message);
+            setCaretPosition(newCaretPosition);
+
+            if (textboxRef.current) {
+                Utils.setCaretPosition(textboxRef.current.getInputBox(), newCaretPosition);
+            }
+        };
+
+        document.addEventListener('paste', handlePaste);
+
+        return () => {
+            document.removeEventListener('paste', handlePaste);
+        };
+    }, []);
+
+    const isSaveDisabled = () => {
+        const {post} = editingPost;
+        const hasAttachments = post && post.file_ids && post.file_ids.length > 0;
+
+        if (hasAttachments) {
+            return !rest.canEditPost;
+        }
+
+        if (editText.trim() !== '') {
+            return !rest.canEditPost;
+        }
+
+        return !rest.canDeletePost;
+    };
+
+    const applyHotkeyMarkdown = (e: React.KeyboardEvent) => {
+        const res = Utils.applyHotkeyMarkdown(e);
+
+        setEditText(res.message);
+        if (textboxRef.current) {
+            Utils.setSelectionRange(
+                textboxRef.current.getInputBox(),
+                res.selectionStart,
+                res.selectionEnd,
+            );
+        }
+    };
+
+    const handleRefocusAndExit = (refocusId: string|null) => {
+        if (refocusId) {
+            setTimeout(() => {
```

Comment:
You are right about this ... but only in conjunction with the second comment about resetting the state. I left the setTimeout in because it was refocussing before the unmount happened. I just wanted to make sure that the textbox is only focussed when everything is cleared up, so we do not accidentally land in a weird state.

I removed that and the state resets after checking it is functional locally.